### PR TITLE
Tests and bugfixes

### PR DESCRIPTION
**Pull request by dstathis**
Fix a couple bugs and add some basic tests

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -6,3 +6,6 @@ options:
     default: 'cassandra'
     description: The docker image to use
     type: string
+  port:
+    default: 9042
```
Comment (balbirthomas) [inline config.yaml:10]: Do you need port to be a config parameter instead of a hardcoded constant in your charm ?

**Review by balbirthomas — APPROVED** [review]

File: config.yaml (hunk 1 above)
Comment (dstathis) [reply config.yaml:10]: Yes this was a requirement

File: src/charm.py [hunk 2]
```diff
@@ -66,17 +75,22 @@ def build_pod_spec(self):
                     "command": [
                         "sh",
                         "-c",
-                        "echo $(cat /etc/hosts | grep cassandra-endpoints | cut -f 1) listen-addr >> /etc/hosts && docker-entrypoint.sh",
+                        "echo $(cat /etc/hosts | grep cassandra-endpoints | cut -f 1) listen-addr"
```
Comment (Abuelodelanada) [inline src/charm.py:78]: If i remember well @jlounder said that we could use: `0.0.0.0`

**Review by Abuelodelanada — APPROVED** [review]
Looks good to me

**Pull request merged**